src/db.rs

```diff
@@ -89,27 +100,46 @@ impl DBStore {
             live_files.iter().map(|f| f.size).sum::<usize>() as f64 / 1e9,
             live_files.iter().map(|f| f.num_entries).sum::<u64>() as f64 / 1e9
         );
+        let mut config = Self::get_config(&db);
+        debug!("DB {:?}", config);
+        if config.format != CURRENT_FORMAT {
```

Comment:
Actually realized this will reindex on downgrades too. I don't think we want it, do we?

I'm thinking of changing `auto_reindex` to some sort of bit flags (how do we deserialize them?):

* Upgrades - `config.format` < `CURRENT_FORMAT`
* Downgrades - `config.format` > `CURRENT_FORMAT`
* Inconsistencies - everything else
* Always - like `-reindex` in bitcoind - forces reindex; perhaps a separate argument?

Default is Upgrades + Inconsistencies

Reply:
Perhaps could be specified as comma-separated list:

* CLI: `--reindex=upgrades,downgrades,inconsistencies`
* config: `reindex = "upgrades,downgrades,inconsistencies"`

Not sure what `bitflags` crate does to deserialize.

Reply:
Let's keep it simple for now - and re-index every time the DB format changes :)

Reply:
I wouldn't mind writing flags but OK. :)